login_test.rb

```diff
@@ -3,53 +3,86 @@
 require 'test/unit'
 include Test::Unit::Assertions
 
-puts "Enter Username: "
-userID = gets
-puts "Enter Password: "
-pass = gets
-
-driver = Selenium::WebDriver.for :edge
-wait = Selenium::WebDriver::Wait.new(:timeout => 10)
-# userID = "ID"
-# pass = "PASS"
-cartItem = "ITEM"
-
-
-driver.navigate.to "https://www.tokopedia.com/"
-
-findLogin = wait.until {driver.find_element(xpath: "//button[@data-testid= 'btnHeaderLogin']")}
-findLogin.click
-
-enterID = wait.until {driver.find_element(xpath: "//input[@data-testid= 'email-phone-input']")}
-enterID.send_keys +userID
-
-proceedToPass = wait.until {driver.find_element(xpath: "//button[@data-testid= 'email-phone-submit']")}
-proceedToPass.click
-
-enterPass = wait.until {driver.find_element(xpath: "//input[@autocomplete= 'current-password']")}
-enterPass.send_keys +pass
-
-login = wait.until {driver.find_element(xpath: "//span[@aria-label= 'login-button']")}
-login.submit
-
-verify = wait.until {driver.find_element(xpath: "//img[@src = 'https://ecs7.tokopedia.net/otp/cotp/ICON_EMAIL_NEW.png']")}
-verify.click
-
-puts "Enter verification code: "
-verificationCode = gets
-
-otp = wait.until {driver.find_element(xpath: "//input[@autocomplete = 'one-time-code']")}
-otp.send_keys +verificationCode	
-
-inputItemName = wait.until {driver.find_element(xpath: "//input[@data-unify= 'Search']")}
-inputItemName.send_keys +cartItem
-
-searchItem = wait.until {driver.find_element(xpath: "//button[@aria-label= 'Tombol pencarian']")}
-searchItem.submit
-
-#findCorrectItem = wait.until {driver.find_element(xpath: "//a[contains(@title= 'IPhone 13 Pro Max')]")}
-#findCorrectItem.click
-
-#addToCart = wait.until {driver.find_element(xpath: "//button[@data-testid = 'pdpBtnNormalPrimary']")}
-#addToCart.click
-
+url = "https://accounts.bukalapak.com/login?"
+
+def get_data
+	puts "Enter email: "
+	user = gets
+	puts "Enter password: "
+	pass = gets
+	puts "Enter item to add to cart"
+	item = gets
+end
+
+#navigate function
+def open_website (web_name)
+	driver = Selenium::WebDriver.for :chrome
+	driver.navigate.to web_name
+end
+
+def wait (wait_time)
+	Selenium::WebDriver::Wait.new(:timeout => wait_time.to_i)
+end
+
+def pathfinder(path, wait_time = 10)
+	driver = Selenium::WebDriver.for :chrome

```

Comment:
driver yang disini sama di line 19 itu 2 hal yang berbeda

biasanya pake instance variable

https://www.geeksforgeeks.org/instance-variables-in-ruby/#:~:text=An%20instance%20variable%20in%20ruby,values%20for%20their%20instance%20variables.

Reply:
mending hapus yang instance variable atau yang di line 19 (global variable?)?

Reply:
contoh gini
```
def A
   @something = 2
   something_else = 2
end

def B
   @something=3
   something_else = 3
 end
 
 @something=1 # something = 1
 something_else  = 1 # something_else = 1
 A # something = 2 dan something_else = 1
 B # something = 3  dan something_else = 1
 ```